docs/generated/stablehlo_passes.md

```diff
@@ -338,6 +338,143 @@ Modules valid for shape refinement must have the following properties:
+    naming conflicts within the module.
+
+This pass can be used in three distinct ways:
+
+**Mode 1: Command-line Usage**
+
+This mode is the simplest, using the `stablehlo-opt` utility with the
+`op-names` (a comma-separated list of operation names) and `version` (an
+integer version number) options. It wraps **all instances** of specified
+operations. The attributes of the newly created `stablehlo.composite`
+operation will be the same as the attributes of the original operation.
+
+**Usage Example:**
+
+```bash
+stablehlo-opt input.mlir --stablehlo-wrap-in-composite=op-names='stablehlo.add,stablehlo.mul' -o output.mlir
+```
+
+**Mode 2: Programmatic Single-Op Wrapping**
+
+This mode provides programmatic control to wrap
+**a specific operation instance** and returns a pointer to the newly
+created `stablehlo.composite` operation.
+
+**Example (C++):**
+
+```cpp
+// To wrap a specific stablehlo.add instance
+
+mlir::stablehlo::AddOp addOp = ...; // The op instanced to be wrapped.
+mlir::ModuleOp module = addOp->getParentOfType<mlir::ModuleOp>();
+mlir::OpBuilder builder(addOp);
+mlir::NamedAttrList attrs = ...; // Attributes to be set on the composite op.
+int32_t version = 0; // Composite version.
+
+mlir::stablehlo::CompositeOp compositeOp = mlir::stablehlo::wrapOperationInComposite(builder, addOp, attrs, version, module);
+addOp.replaceAllUsesWith(compositeOp);
+```
+
+**Mode 3: Programmatic Module-Wide Wrapping with customized Attribute Handling**
+
+This mode extends programmatic wrapping to the entire module, offering
+fine-grained control over which operations are wrapped and their attributes.
+This is achieved by using the `createStablehloWrapInCompositePass` API,
+which takes an `CompositeAttributeProviderMap` as an argument.
+
+The `CompositeAttributeProviderMap` is a map that dictates which operations
+should be considered for wrapping and how their attributes should be
+handled. Its semantics are as follows:
+
+- **Keys (mlir::TypeID):** `TypeID` of an MLIR operation. If an operation's
+    `TypeID` matches a key in the map, it becomes a candidate for wrapping.
+- **Values (Lambda Functions):** Lambda function of type
+    `std::function<std::optional<NamedAttrList>(Operation*)>`. This function
+    is applied to each candidate operation.
+    - **Input:** An `mlir::Operation*`, which is an instance of the
+      operation type corresponding to the `TypeID` key.
+    - **Return Value:** An `std::optional<NamedAttrList>`.
+      - If the lambda returns a `NamedAttrList` (wrapped in
+        `std::optional`), the operation is wrapped in a
+        `stablehlo::composite` operation, and the returned attributes are
+        used to set the composite's attributes.
+      - If the lambda returns `std::nullopt`, the operation is **not**
+        wrapped. This allows for selective wrapping based on custom
+        criteria.
+
+**Example (C++):**
+
+```cpp
+
+// ... inside a pass or function ...
```

Comment:
Remove code comment, this is just a pass config so it can be anywhere technically. Probably _shouldn't_ be inside another pass